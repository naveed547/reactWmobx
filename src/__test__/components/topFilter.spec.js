import React from "react";
//import {mount} from "enzyme";
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import TopFilter from "../../components/topFilter";
//import initialState from '../../reducers/initialState';
//import repoMock from "../data/mock";

describe("TopFilter", () => {
    it('TopFilter component renders correctly', () => {
        const props = {
            sort: jest.fn(),
            toggle: jest.fn(),
            issuesCount: {
              open_count: 1000,
              closed_count: 500
            },
            sortFilter: 'Oldest',
            toggleFilter: 'open'
        };
        const rendered = renderer.create(
            <MemoryRouter>
                <TopFilter {...props} />
            </MemoryRouter>
        );
        expect(rendered.toJSON()).toMatchSnapshot();
    });
    /* it("displays the userRepoSearch component when it has fetched users", () => {
        const props = {
            loadRepos: jest.fn(),
            repoDetails: {
                ...initialState.repoDetails,
                repoData:   repoMock.userRepos
            }
        };

        const container = mount(
            <MemoryRouter>
                <UserRepoSearch {...props} />
            </MemoryRouter>
        );
        expect(container.find('.user-name').text()).toEqual(props.repoDetails.userName);
        expect(container.find('.repo-list li').length).toEqual(props.repoDetails.repoData.length);
    });
  
    it("does not display the userRepoSearch when ther are no users", () => {
        const props = {
            loadRepos: jest.fn(),
            repoDetails: {
                userName: ' ',
		        currentRepo: ' ',
                repoData:   []
            }
        };
    
        const container = mount(
            <MemoryRouter>
                <UserRepoSearch {...props} />
            </MemoryRouter>
        );
        expect(container.find('.user-name').text()).toEqual(' ');
        expect(container.find('.repo-list li').length).toEqual(0);
    }); */
  });
  