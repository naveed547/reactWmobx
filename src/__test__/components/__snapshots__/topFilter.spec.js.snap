// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopFilter TopFilter component renders correctly 1`] = `
<div
  className="topFilter row"
>
  <div
    className="col text-center"
  >
    <button
      className="btn btn-secondary"
      onClick={[Function]}
      type="button"
    >
      <i
        className="fa fa-exclamation-circle text-danger"
      />
       
      1000
       Open
    </button>
    <button
      className="btn btn-outline-secondary ml-1"
      onClick={[Function]}
      type="button"
    >
      <i
        className="fa fa-check text-success"
      />
       
      500
       Closed
    </button>
  </div>
  <div
    className="col text-right"
  >
    <select
      className="form-control"
      onChange={[Function]}
      value="Oldest"
    >
      <option>
        Newest
      </option>
      <option>
        Oldest
      </option>
      <option>
        Most Commented
      </option>
      <option>
        Least Commented
      </option>
    </select>
  </div>
</div>
`;
